src/config.c

```diff
@@ -0,0 +1,199 @@
+        },
+    .cf_bucket_size =
+        {
+            .value = 2,
+            .min = 1,
+            .max = 255,
+        },
+    .cf_initial_size =
+        {
+            .value = 1024,
+            .min = 4,
+            .max = 1LL << 30,
+        },
+    .cf_max_iterations =
+        {
+            .value = 20,
+            .min = 1,
+            .max = 65535,
+        },
+    .cf_expansion_factor =
+        {
+            .value = 1,
+            .min = 0,
+            .max = 32768,
+        },
+    .cf_max_expansions =
+        {
+            .value = 32,
+            .min = 1,
+            .max = 65535,
+        },
+};
+
+static int setFloatValue(const char *name, RedisModuleString *value, void *privdata,
+                         RedisModuleString **err) {
+    RM_ConfigFloat *config = privdata;
+    double *val = &config->value;
+    double new_val;
+    if (RedisModule_StringToDouble(value, &new_val) != REDISMODULE_OK) {
+        *err = RedisModule_CreateStringPrintf(NULL, "Invalid value for `%s`", name);
+        return REDISMODULE_ERR;
+    }
+    if (new_val < config->min || new_val > config->max) {
+        *err = RedisModule_CreateStringPrintf(NULL, "Value for `%s` must be in the range [%f, %f]",
+                                              name, config->min, config->max);
+        return REDISMODULE_ERR;
+    }
+    *val = new_val;
+    return REDISMODULE_OK;
+}
+static RedisModuleString *getFloatValue(const char *name, void *privdata) {
+    return RedisModule_CreateStringPrintf(NULL, "%f", ((RM_ConfigFloat *)privdata)->value);
+}
+
+static int setNumericValue(const char *name, RedisModuleString *value, void *privdata,
+                           RedisModuleString **err) {
+    RM_ConfigNumeric *config = privdata;
+    long long *val = &config->value;
+    long long new_val;
+    if (RedisModule_StringToLongLong(value, &new_val) != REDISMODULE_OK) {
+        *err = RedisModule_CreateStringPrintf(NULL, "Invalid value for `%s`", name);
+        return REDISMODULE_ERR;
+    }
+    if (new_val < config->min || new_val > config->max) {
+        *err =
+            RedisModule_CreateStringPrintf(NULL, "Value for `%s` must be in the range [%lld, %lld]",
+                                           name, config->min, config->max);
+        return REDISMODULE_ERR;
+    }
+    *val = new_val;
+
+    if (strcasecmp(name, "cf-bucket-size") == 0) {
+        rm_config.cf_initial_size.min = new_val * 2;
+    } else if (strcasecmp(name, "cf-initial-size") == 0) {
+        rm_config.cf_bucket_size.max = new_val / 2 < 255 ? new_val / 2 : 255;
+    }
+
+    return REDISMODULE_OK;
+}
+static RedisModuleString *getNumericValue(const char *name, void *privdata) {
+    return RedisModule_CreateStringPrintf(NULL, "%zu", *((size_t *)privdata));
+}
+
+typedef struct {
+    const char *name;
+    const char *defVal;
+    RedisModuleConfigSetStringFunc setValue;
+    RedisModuleConfigGetStringFunc getValue;
+    void *privdata;
```

Comment:
Per the CONFIG spec, all configs have `privdata` (private data) that Redis doesn't use. It simply stores it to be passed to the `setValue` and `getValue` functions.

In this case, we use it to store a pointer to the module-side configuration variable so that we can update it whenever `CONFIG SET` is called, and read from it whenever `CONFIG GET` is called. We do not store the config variable on the Redis-side as if we did that then we would need to make an external call every time we wanted to update or read the variable within the module.